### ISSUE TITLE
Feature specification content on the website

### ISSUE THREAD

**Issue by strogonoff** [riboseinc/metanorma.com#5]

**Comment by strogonoff** [comment]
Upstream issues:

* https://github.com/riboseinc/jekyll-theme-open-project-helpers/issues/9
* https://github.com/riboseinc/jekyll-theme-open-project/issues/38

**Comment by strogonoff** [comment]
Current implementation integrates model diagrams (screenshot). To do before release:

* Fix specification layout to match software docs (with expandable narrow-viewport-friendly navigation bar)
* Make build work with Metanorma-based specs in addition to PNG diagrams

<img width="1416" alt="screen shot 2018-11-25 at 3 04 35 pm" src="https://user-images.githubusercontent.com/292959/48976495-cd646d80-f0c3-11e8-9885-aca541a5f77d.png">

**Comment by ronaldtse** [comment]
I wonder if we should directly incorporate PlantUML into site generation rather than relying on PNG files?

**Comment by strogonoff** [comment]
Sure, could be another plugin. This one is convenient in that if you switch to another diagram generator it won’t break project sites downstream, and that it takes any PNGs, not just a UML spec.

**Comment by ronaldtse** [comment]
Yes indeed. Maybe that is desirable. Let's hold off PlantUML for now.

**Comment by strogonoff** [comment]
Yeah seems pragmatic to hold off PlantUML for now since handling its dependencies (or containerizing the whole thing) complicates things. Adding the Metanorma engine next, there’re some challenges there too. PNG illustration engine to fix #34 is working at the moment, pending deployment: http://www.metanorma.comspecs/metanorma-model-gb/

<img width="1322" alt="screen shot 2018-11-28 at 9 03 00 am" src="https://user-images.githubusercontent.com/292959/49121719-6ece0800-f2ec-11e8-8ae7-eb6dcd70d13c.png">

**Comment by ronaldtse** [comment]
BTW it is easy to build Metanorma documents now using Travis, you can see the Travis CI file here https://github.com/riboseinc/ogc-tcpnp. 

So it should be really easy to have a Jekyll plugin that renders Metanorma.

**Comment by strogonoff** [comment]
To build PlantUML sources into SVG, rather than rely on images being part of model repo, it should be possible to call these methods: https://github.com/nus/gitbucket-plantuml-plugin/blob/master/src/main/scala/com/yotaichino/gitbucket/plugins/plantuml/PlantUMLUtils.scala#L12

**Comment by ronaldtse** [comment]
You can consider the `asciidoctor-diagram` gem that allows this?
